doc/developers/contributing.rst

```diff
@@ -1325,6 +1327,40 @@ interface might be that you want to use it together with model evaluation and
 selection tools such as :class:`model_selection.GridSearchCV` and
 :class:`pipeline.Pipeline`.
 
+Setting `generate_only=True` returns a generator that yields (estimator, check)
+tuples where the check can be called independently from each other, i.e.
+`check(estimator)`. This allows all checks to be run independently and report
+the checks that are failing. In pytest, we can parameterize the check as
+follows::
+
+  import pytest
+  from sklearn.utils.estimator_checks import check_estimator
+  from sklearn.utils.estimator_checks import set_check_estimator_ids
+  from sklearn.linear_model import LogisticRegression
+
+  @pytest.mark.parametrize(
+      'estimator, check',
+      check_estimator(LogisticRegression, generate_only=True)
+      ids=set_check_estimator_ids)
+  def test_sklearn_compatible_estimator(estimator, check):
+      check(estimator)
+
+When testing more than one estimator, python provides a `chain.from_iterable`
```

Comment:
I'm not super convinced of the benefit of this example. This is basically advertising for `chain`.

I'd rather have an example explaining how using `-k` on the ids can help selecting a specific check. Also, the `set_check_estimator_ids` should be briefly explained.